docs/tutorial/part-zero/index.md

```diff
@@ -0,0 +1,23 @@
+---
+title: Programming Basics (Part Zero)
```

Comment:
I think this is less of a "programming basics" and more of a "development environment" primer/introduction. Would you agree? Linking back to [this comment](https://github.com/gatsbyjs/gatsby/pull/4371#issuecomment-370665024), I see this being more of a structured walkthrough (with a targeted intro to concepts as they arise), rather than a listing of concepts.